akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -476,6 +476,86 @@ private[akka] final case class Conflate[In, Out](seed: In ⇒ Out, aggregate: (O
 /**
  * INTERNAL API
  */
+private[akka] final case class AggregateWeighted[In, Out](max: Long, costFn: In ⇒ Long, seed: In ⇒ Out,
+                                                          aggregate: (Out, In) ⇒ Out,
+                                                          decider: Supervision.Decider) extends DetachedStage[In, Out] {
```

Comment:
Can you reimplement this as a `GraphStage`? The DetachedStage will be phased out eventually. The differences are not much (explained in the migration guide with example).